shared/reducers/chat2.js

```diff
@@ -243,9 +243,13 @@ const messageMapReducer = (messageMap, action, pendingOutboxToOrdinal) => {
         return message.set('downloadPath', path)
       })
     case Chat2Gen.metasReceived:
-      return action.payload.clearExistingMessages
-        ? messageMap.filter((_, k) => k !== Constants.pendingConversationIDKey)
-        : messageMap
+      const existingPending = messageMap.get(Constants.pendingConversationIDKey)
+      if (action.payload.clearExistingMessages) {
+        return existingPending
+          ? messageMap.clear().set(Constants.pendingConversationIDKey, existingPending)
```

Comment:
probably safe to call set even if existingPending is undefined

Reply:
Flow was bugging me about it